### PR TITLE
[Feature] Support `LaserMix` augmentation

### PR DESCRIPTION
**Pull request by Xiangxu-0103**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet3d/datasets/transforms/transforms_3d.py [hunk 1]
```diff
@@ -2352,3 +2353,142 @@ def transform(self, input_dict: dict) -> dict:
...
+        num_areas = np.random.choice(self.num_areas, size=1)[0]
+        angle_list = np.linspace(self.pitch_angles[1], self.pitch_angles[0],
+                                 num_areas + 1)
+        out_points = []
+        out_pts_semantic_mask = []
+        for i in range(num_areas):
+            # convert angle to radian
+            start_angle = angle_list[i + 1] / 180 * np.pi
+            end_angle = angle_list[i] / 180 * np.pi
+            if i % 2 == 0:  # pick from original point cloud
```
Comment (ldkong1205) [inline mmdet3d/datasets/transforms/transforms_3d.py:2442]: Can we have a hyperparameter to determine the mixing probability? 0.5 might not be the optimal choice based on my previous observations. Thanks!

Comment (Xiangxu-0103) [reply mmdet3d/datasets/transforms/transforms_3d.py:2445]: Added

**Comment by ZwwWayne** [comment]
Hi @Xiangxu-0103 ,
Thanks for your kind PR. It seems that there are conflicts between your PR and the targeted branch. Could you solve the conflicts so that we could eventually merge this PR?

You can find more details in the place shown as below:
image

**Comment by Xiangxu-0103** [comment]
> Hi @Xiangxu-0103 , Thanks for your kind PR. It seems that there are conflicts between your PR and the targeted branch. Could you solve the conflicts so that we could eventually merge this PR?
> 
> You can find more details in the place shown as below: image

**Comment by Xiangxu-0103** [comment]
> Hi @Xiangxu-0103 , Thanks for your kind PR. It seems that there are conflicts between your PR and the targeted branch. Could you solve the conflicts so that we could eventually merge this PR?
> 
> You can find more details in the place shown as below: image

resolved.

**Review by JingweiZhang12 — APPROVED** [review]

**Review by ZCMax — APPROVED** [review]

**Pull request merged**